### ISSUE TITLE
iOS build failing after upgrading to expo 46 & doing prebuild

### ISSUE THREAD

**Issue by KFrysztak** [expo/eas-cli#1255]
### Build/Submit details page URL

https://expo.dev/accounts/family-app/projects/Lifebonus/builds/b61cb9ac-d8b3-415e-8973-a324626548a1

### Summary

After upgrading to expo 46 EAS build for iOS is failing constantly. 
This is happening both when triggered from our github actions + locally.

### Managed or bare?

bare

### Environment

```
expo-env-info 1.0.5 environment info:
    System:
      OS: Windows 10 10.0.22000
    Binaries:
      Node: 16.15.0 - C:\Program Files\nodejs\node.EXE
      npm: 8.16.0 - C:\Program Files\nodejs\npm.CMD
    IDEs:
      Android Studio: AI-212.5712.43.2112.8512546
    npmPackages:
      @expo/metro-config: ^0.3.21 => 0.3.21
      expo: ^46.0.0 => 46.0.2
      react: 18.0.0 => 18.0.0
      react-dom: 18.0.0 => 18.0.0
      react-native: 0.69.3 => 0.69.3
      react-native-web: ~0.18.7 => 0.18.7
    Expo Workflow: bare
```

Expo doctor is clear

### Error output

```
› Executing react-native Pods/FBReactNativeSpec » [CP-User] Generate Specs
▸ ** ARCHIVE FAILED **
▸ The following build commands failed:
▸ 	CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler (in target 'ExpoModulesCore' from project 'Pods')
▸ 	PhaseScriptExecution [CP-User]\ Generate\ Specs /Users/expo/Library/Developer/Xcode/DerivedData/Lifebonus-ckyaviiwysbzpggaynlnsatcgqmi/Build/Intermediates.noindex/ArchiveIntermediates/Lifebonus/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/FBReactNativeSpec.build/Script-46EB2E000179A0.sh (in target 'FBReactNativeSpec' from project 'Pods')
▸ (2 failures)
2022-08-08 12:31:58.166 xcodebuild[4219:12713] Requested but did not find extension point with identifier Xcode.IDEKit.ExtensionSentinelHostApplications for extension Xcode.DebuggerFoundation.AppExtensionHosts.watchOS of plug-in com.apple.dt.IDEWatchSupportCore
2022-08-08 12:31:58.166 xcodebuild[4219:12713] Requested but did not find extension point with identifier Xcode.IDEKit.ExtensionPointIdentifierToBundleIdentifier for extension Xcode.DebuggerFoundation.AppExtensionToBundleIdentifierMap.watchOS of plug-in com.apple.dt.IDEWatchSupportCore
2022-08-08 12:31:58.224 xcodebuild[4219:12713] XType: failed to connect - Error Domain=NSCocoaErrorDomain Code=4099 "The connection to service named com.apple.fonts was invalidated: failed at lookup with error 3 - No such process." UserInfo={NSDebugDescription=The connection to service named com.apple.fonts was invalidated: failed at lookup with error 3 - No such process.}
2022-08-08 12:31:58.224 xcodebuild[4219:12713] Font server protocol version mismatch (expected:5 got:0), falling back to local fonts
2022-08-08 12:31:58.224 xcodebuild[4219:12713] XType: unable to make a connection to the font daemon!
2022-08-08 12:31:58.224 xcodebuild[4219:12713] XType: XTFontStaticRegistry is enabled as fontd is not available.
** ARCHIVE FAILED **
The following build commands failed:
	CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler (in target 'ExpoModulesCore' from project 'Pods')
	PhaseScriptExecution [CP-User]\ Generate\ Specs /Users/expo/Library/Developer/Xcode/DerivedData/Lifebonus-ckyaviiwysbzpggaynlnsatcgqmi/Build/Intermediates.noindex/ArchiveIntermediates/Lifebonus/IntermediateBuildFilesPath/Pods.build/Release-iphoneos/FBReactNativeSpec.build/Script-46EB2E000179A0.sh (in target 'FBReactNativeSpec' from project 'Pods')
(2 failures)
Exit status: 65
```

### Reproducible demo or steps to reproduce from a blank project

```
  "dependencies": {
    "@apollo/client": "^3.6.9",
    "@expo/metro-config": "^0.3.21",
    "@expo/vector-icons": "^13.0.0",
    "@react-native-async-storage/async-storage": "~1.17.7",
    "@react-navigation/bottom-tabs": "^6.3.2",
    "@react-navigation/native": "^6.0.11",
    "@react-navigation/stack": "^6.2.2",
    "@sentry/react-native": "^4.1.3",
    "axios": "^0.27.2",
    "babel-plugin-inline-dotenv": "^1.7.0",
    "eslint": "^8.21.0",
    "expo": "^46.0.0",
    "expo-application": "~4.2.2",
    "expo-constants": "~13.2.3",
    "expo-dev-client": "~1.1.1",
    "expo-device": "~4.3.0",
    "expo-localization": "~13.1.0",
    "expo-splash-screen": "~0.16.1",
    "expo-status-bar": "~1.4.0",
    "expo-updates": "~0.14.3",
    "graphql": "^16.5.0",
    "i18next": "^21.8.16",
    "jest": "^26.6.3",
    "jest-expo": "^46.0.0",
    "react": "18.0.0",
    "react-dom": "18.0.0",
    "react-i18next": "^11.18.3",
    "react-native": "0.69.3",
    "react-native-gesture-handler": "~2.5.0",
    "react-native-safe-area-context": "4.3.1",
    "react-native-screens": "~3.15.0",
    "react-native-svg": "12.3.0",
    "react-native-svg-transformer": "^1.0.0",
    "react-native-svg-web": "^1.0.9",
    "react-native-web": "~0.18.7",
    "sentry-expo": "~5.0.0"
  },
  "devDependencies": {
    "@babel/core": "^7.18.6",
    "@types/jest": "^28.1.6",
    "@types/react": "~18.0.0",
    "@types/react-native": "~0.69.1",
    "@types/react-test-renderer": "^18.0.0",
    "@typescript-eslint/eslint-plugin": "^5.32.0",
    "@typescript-eslint/parser": "^5.32.0",
    "babel-plugin-inline-react-svg": "^2.0.1",
    "eslint-config-airbnb": "^19.0.4",
    "eslint-config-airbnb-typescript": "^17.0.0",
    "eslint-plugin-jsx-a11y": "^6.6.1",
    "eslint-plugin-react": "^7.30.1",
    "react-test-renderer": "18.0.0",
    "typescript": "^4.6.3"
  },
```

**Comment by dsokal** [comment]
Hi,

Unfortunately, I cannot reproduce this issue. I initialized a new project with `npx create-expo-app --template blank@sdk-46` and I was able to build an iOS build with EAS. I also installed all of the dependencies you shared in the snippet. This also didn't make the build fail. Can you please prepare a minimal reproducible example of this issue? https://stackoverflow.com/help/minimal-reproducible-example

FWIW I think the error you shared is not actually the culprit. I checked the Xcode logs for your build (available under the regular build logs) and the error seems to be:
```
Node found at: /usr/local/bin/node
/Users/expo/workingdir/build/mobile-app/ios/Pods/../../node_modules/react-native/React/FBReactNativeSpec/../../scripts/react_native_pods_utils/script_phases.sh: line 83: /usr/local/bin/node: No such file or directory
```

I think you might have made some mistake when upgrading your bare project to the new SDK.

**Comment by KFrysztak** [comment]
I did the other way around, first I have updated expo and only then I am trying to go bare.

And then when I am building without ios folder all is good, expo can prebuild without any problems on expo.dev. But when we are trying to prebuild on our side (different machines, different varsions of all tools) then it no longer builds. 

Is there maybe an option to get somehow the result of prebuild that is run on expo.dev pipeline? As we were not changing it anyhow.
